airbyte-integrations/connectors/source-mysql-v2/src/test/kotlin/io/airbyte/integrations/source/mysql/MysqlCursorBasedIntegrationTest.kt

```diff
@@ -0,0 +1,129 @@
+        val recordMessageFromRun2: List<AirbyteRecordMessage> = run2.records()
+        assertEquals(recordMessageFromRun2.size, 1)
+    }
+
+    companion object {
+        val log = KotlinLogging.logger {}
+        lateinit var dbContainer: MySQLContainer<*>
+
+        fun config(): MysqlSourceConfigurationJsonObject =
+            MysqlContainerFactory.config(dbContainer).apply {
+                setCursorMethodValue(UserDefinedCursor)
+            }
+
+        val connectionFactory: JdbcConnectionFactory by lazy {
+            JdbcConnectionFactory(MysqlSourceConfigurationFactory().make(config()))
+        }
+
+        val configuredCatalog: ConfiguredAirbyteCatalog = run {
+            val desc = StreamDescriptor().withName("tbl").withNamespace("test")
+            val discoveredStream =
+                DiscoveredStream(
+                    id = StreamIdentifier.Companion.from(desc),
+                    columns = listOf(Field("k", IntFieldType), Field("v", StringFieldType)),
+                    primaryKeyColumnIDs = listOf(listOf("k")),
+                )
+            val stream: AirbyteStream = JdbcAirbyteStreamFactory().createGlobal(discoveredStream)
+            val configuredStream: ConfiguredAirbyteStream =
+                CatalogHelpers.toDefaultConfiguredStream(stream)
+                    .withSyncMode(SyncMode.INCREMENTAL)
+                    .withPrimaryKey(discoveredStream.primaryKeyColumnIDs)
+                    .withCursorField(listOf("k"))
+            ConfiguredAirbyteCatalog().withStreams(listOf(configuredStream))
+        }
+
+        @JvmStatic
+        @BeforeAll
+        @Timeout(value = 300)
+        fun startAndProvisionTestContainer() {
+            dbContainer =
+                MysqlContainerFactory.exclusive(
+                    imageName = "mysql:8.0",
+                    MysqlContainerFactory.WithNetwork,
+                )
+            provisionTestContainer(dbContainer, connectionFactory)
+        }
+
+        fun provisionTestContainer(
+            targetContainer: MySQLContainer<*>,
+            targetConnectionFactory: JdbcConnectionFactory
+        ) {
+            val grant =
+                "GRANT SELECT, RELOAD, SHOW DATABASES, REPLICATION SLAVE, REPLICATION CLIENT " +
+                    "ON *.* TO '${targetContainer.username}'@'%';"
+            targetContainer.execAsRoot(grant)
+            targetContainer.execAsRoot("FLUSH PRIVILEGES;")
+
+            targetConnectionFactory.get().use { connection: Connection ->
+                connection.isReadOnly = false
+                connection.createStatement().use { stmt: Statement ->
+                    stmt.execute("CREATE TABLE test.tbl(k INT PRIMARY KEY, v VARCHAR(80))")
+                }
+                connection.createStatement().use { stmt: Statement ->
+                    stmt.execute("INSERT INTO test.tbl (k, v) VALUES (1, 'foo'), (2, 'bar')")
+                }
+            }
```

Comment:
you might as well move this into the test method itself
if you're going to use a shared container (you should) then you'll have to name the table something more unique or there will be a risk of collision with tables created in other tests

Reply:
for now i just lightly refactored it to create different table names (with same content) for each test, since now they share the same dbcontainer. If future use case evolves we can revisit.